src/NServiceBus.Core/Pipeline/Outgoing/InspectOutgoingMessageBehavior.cs

```diff
@@ -0,0 +1,16 @@
+﻿namespace NServiceBus
+{
+    using System;
+    using System.Threading.Tasks;
+    using NServiceBus.Pipeline;
+
+    class InspectOutgoingMessageBehavior : Behavior<IOutgoingPhysicalMessageContext>
```

Comment:
Wouldn't it be simpler instead of creating yet another behavior to do the following?

* Implement in `ImmediateDispatchTerminator` or
* Implement in the `TransportOperations` collection constructor on all operations

No additional behavior needed and works for batching and immediate dispatch.